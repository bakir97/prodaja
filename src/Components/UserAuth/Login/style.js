export default {
  root: { height: "100vh", padding: "0 1rem" },
  button: {
    fontFamily: "Kalam",
    fontSize: "1.5rem",
    paddingBottom: 0
  }
};
