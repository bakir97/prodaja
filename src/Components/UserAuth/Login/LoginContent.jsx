import React from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import TextComponent from "../../../Layout/FormElements/Text";
import { Button, Grid } from "@material-ui/core";
import { Field } from "redux-form";
import style from "./style";
export default ({ loading, errorLogin }) => {
  return (
    <Grid
      direction="column"
      container
      style={style.root}
      justify="center"
      alignItems="center"
    >
      <Field
        name="username"
        serverError={errorLogin.username && errorLogin.username}
        label="Username"
        component={TextComponent}
      />
      <Field
        serverError={errorLogin.password && errorLogin.password}
        name="password"
        label="Password"
        component={TextComponent}
        type="password"
      />
      {loading ? (
        <CircularProgress />
      ) : (
        <Button style={style.button} type="submit">
          Login
        </Button>
      )}
    </Grid>
  );
};
