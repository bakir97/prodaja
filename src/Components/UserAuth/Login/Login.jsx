import React, { Component } from "react";
import { connect } from "react-redux";
import { login } from "../../../redux/actions/loginAction";
import LoginContent from "./LoginContent";
import { reduxForm } from "redux-form";

export class Login extends Component {
  submit = e => {
    this.props.login(e);
  };
  render() {
    return (
      <form onSubmit={this.props.handleSubmit(this.submit)}>
        <LoginContent {...this.props} />
      </form>
    );
  }
}

const mapStateToProps = state => ({
  account: state.login.account,
  errorLogin: state.login.errorLogin,
  loading: state.login.loading
});

const mapDispatchToProps = {
  login
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(reduxForm({ form: "login" })(Login));
