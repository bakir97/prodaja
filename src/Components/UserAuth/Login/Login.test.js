import { Login } from "./Login";
import LoginContent from "./LoginContent";
describe("Login", () => {
  const params = {
    login: jest.fn(),
    handleSubmit: jest.fn()
  };

  const wrapper = shallow(<Login {...params} />);
  it("should call submit", () => {
    wrapper.instance().submit({ name: "Test" });
    expect(params.login).toBeCalledWith({ name: "Test" });
  });
});

describe("LoginContent", () => {
  const wrapper = shallow(<LoginContent errorLogin={{}} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
