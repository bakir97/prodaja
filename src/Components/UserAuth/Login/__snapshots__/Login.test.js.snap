// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContent should render 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  style={
    Object {
      "height": "100vh",
      "padding": "0 1rem",
    }
  }
>
  <Field
    component={[Function]}
    label="Username"
    name="username"
  />
  <Field
    component={[Function]}
    label="Password"
    name="password"
    type="password"
  />
  <WithStyles(Button)
    style={
      Object {
        "fontFamily": "Kalam",
        "fontSize": "1.5rem",
        "paddingBottom": 0,
      }
    }
    type="submit"
  >
    Login
  </WithStyles(Button)>
</WithStyles(Grid)>
`;
